### PR TITLE
Update pygrib method to account for 37 levels

### PR DESCRIPTION
**Pull request by LinlinCui-NOAA**
### Description
- This PR updated `pygrib` method in the `gdas_utility.py` script to account for 37 pressure levels. Specifically, it combines two dataarray along the dimension `level` when `levelType` is `isobaricInhPa` and `num_levels` is 37.
- Fixed a bug in downloading data from nomads server.
- Deleted hard-coded environment variables and added an argument to let the user specify the path to credential files if environment variables are not available

### Linked Issues
- Closes https://github.com/NOAA-EMC/graphcast/issues/30

### Blocking Dependencies

None

### Anticipated Changes
None
### Input data
- [ ] No changes are expected to input data.
- [x] Changes are expected to input data:
  - [x] New input data.
  - [x] Updated input data.

### Needed libraries

None

### CODE REVIEW

File: NCEP/gdas_utility.py [hunk 1]
```diff
@@ -39,19 +39,24 @@ def __init__(self, start_datetime, end_datetime, num_pressure_levels=13, downloa
         if self.download_source == 's3':
             # Initialize the S3 client
             # Specify the path to your custom AWS credentials file
-            custom_credentials_file = '/contrib/Sadegh.Tabas/.aws/credentials'
+            #custom_credentials_file = '/contrib/Sadegh.Tabas/.aws/credentials'
```
Comment (SadeghTabas-NOAA) [inline NCEP/gdas_utility.py:42]: @LinlinCui-NOAA could you please return these, if we want to import credentials then we need to update job cards as well which is not needed. Please just return the commands for this as it was before. Thanks

Comment (SadeghTabas-NOAA) [reply NCEP/gdas_utility.py:42]: Also, I was wondering if there was any issue with the "NOMADS" option. Thanks

**Review by SadeghTabas-NOAA — APPROVED** [review]

**Comment by SadeghTabas-NOAA** [comment]
@LinlinCui-NOAA let me test this on cloud before merging

**Comment by SadeghTabas-NOAA** [comment]
@LinlinCui-NOAA when we run gdas_utility with pygrib method, at the end when the ICs created we should remove `noaa-ncepdev-none-ca-ufs-cpldcld_37` but it doesn't, this cause issues for realtime running, please fix it

**Comment by LinlinCui-NOAA** [comment]
> @LinlinCui-NOAA when we run gdas_utility with pygrib method, at the end when the ICs created we should remove `noaa-ncepdev-none-ca-ufs-cpldcld_37` but it doesn't, this cause issues for realtime running, please fix it

@SadeghTabas-NOAA The script has this feature:
https://github.com/LinlinCui-NOAA/graphcast/blob/f5a2d47cc695d6991ba417416d77f123842b8012/NCEP/gdas_utility.py#L467

And when I tested on Hera, it deleted the folder `ncepdev-none-ca-ufs-cpldcld_37`.

**Pull request merged**